MidasCivil_Adapter/Convert/ToBHoM/Loads/ToBarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            List<double> positions = new List<double>();
+            List<double> temperatureList = new List<double>();
+            DifferentialTemperatureLoadDirection loadDirection = new DifferentialTemperatureLoadDirection();
+            double depth = new double();
+            List<Bar> bhomAssociatedBars = new List<Bar>();
+            Bar bhomAssociatedBar = new Bar();
+            if (matchingBar.Contains(' '))
+            {
+                string[] matchingBars = matchingBar.Split(' ');
+                foreach (string matchBar in matchingBars)
+                    barDictionary.TryGetValue(matchBar, out bhomAssociatedBar);
+                bhomAssociatedBars.Add(bhomAssociatedBar);
+            }
+            else
+            {
+                barDictionary.TryGetValue(matchingBar, out bhomAssociatedBar);
+                bhomAssociatedBars.Add(bhomAssociatedBar);
+            }
+            for (int j = 1; j < temperatureLoad.Count; j++)
+            {
+                string[] delimitted = temperatureLoad[j].Split(',');
+                double absPositionTop = double.Parse(delimitted[4].Trim());
+                double absPositionBot = double.Parse(delimitted[6].Trim());
+
+                if (!positions.Contains(absPositionTop))
+                {
+                    positions.Add(absPositionTop);
+                    temperatureList.Add(double.Parse(delimitted[5].Trim()).DeltaTemperatureToSI(temperatureUnit));
+                }
+                if (!positions.Contains(absPositionBot))
+                {
+                    positions.Add(absPositionBot);
+                    temperatureList.Add(double.Parse(delimitted[7].Trim()).DeltaTemperatureToSI(temperatureUnit));
+                }
+            }
+            if (temperatureLoad[0].Contains("LY"))
+            {
+                loadDirection = DifferentialTemperatureLoadDirection.LocalY;
+                depth = bhomAssociatedBars[0].SectionProperty.Vpy + bhomAssociatedBars[0].SectionProperty.Vy;
+            }
+            else
+            {
+                loadDirection = DifferentialTemperatureLoadDirection.LocalZ;
+                depth = bhomAssociatedBars[0].SectionProperty.Vpz + bhomAssociatedBars[0].SectionProperty.Vz;
+            }
+            List<double> normalisedPositions = new List<double>();
+
+            foreach (double position in positions)
+            {
+                double normalisedPosition = position / depth;
+                if ((1 - normalisedPosition) < 0.02)
+                {
+                    if((1-normalisedPosition>0.001))
+                    {
+                        Compute.RecordWarning("The normalised top position of temperature profile is between 0.981 to 0.999 and has been assumed as 1");
+                    }
+                    normalisedPosition = 1;
+                }
```

Comment:
The second `if` statement is unnecessary and only raises the warning when the normalised position is 0.999 or above. Whereas the warning relates to the first `if` statement too.
```suggestion
                if ((1 - normalisedPosition) < 0.02)
                {
                        Compute.RecordWarning("The normalised top position of temperature profile is between 0.981 to 1.000 and has been assumed as 1");
                         normalisedPosition = 1;
                    }
```